packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -142,6 +143,16 @@ function extractManifestOutput(output: string): string {
   return stripAnsi(output.substring(start, end + 1));
 }
 
+export function populateMetadata(angularVersion?: string): Metadata {
+  const packageJson = JSON.parse(readFileSync(`${__dirname}/../package.json`, "utf-8"));
+  return {
+    adapterNpmPackageName: packageJson.name.replace(/([@])/g, "\\$1"), // escape @ for yamlStringify
+    adapterVersion: packageJson.version,
+    framework: "angular",
+    frameworkVersion: angularVersion ? angularVersion : "",
```

Comment:
Why is angularVersion nullable? When would it not be passed in? Should we be reading this from package.json or package-lock.json?

Reply:
`angularVersion` is the user's angular version passed from framework buildpack with `process.env.FRAMWORK_VERSION`, so is nullable. But since this should always exists, should we be asserting not null instead of defining metadata as having a nullable frameworkVersion?

Reply:
Yeah that sounds reasonable to me!